### PR TITLE
OCRBenchv2 Task 추가

### PR DESCRIPTION
**Pull request by seunggu-upstage**
OCRBenchv2 와 몇 가지 기능 수정 및 추가 했습니다.

- docvision.py에서 phi3의 tokenizer를 사용할 경우, encode 함수를 통해 eos token id를 가져오기에 실패하였습니다. 명시적으로 token id를 가져오는 함수로 수정하여 해당 에러를 방지하였습니다.
- Ablation 모델의 실험을 위해 modeling_phi3.py 스크립트를 추가하였습니다.
- OCRbenchv2의 english subtask만 점수 뽑도록 추가하였습니다.
    - *_metric 스크립트는 OCRBenchv2[https://github.com/Yuliang-Liu/MultimodalOCR/tree/main/OCRBench_v2] 의 레포에서 직접 가져왔습니다
    - lmms-eval repo에서 동작할 수 있도록 세 가지 스크립트를 별도로 작성해주었는데요,
        - ocrbenchv2.yaml; Task의 기본 동작들을 정의하는 파일입니다. HF dataset 경로, task 이름, 데이터 처리, metric 계산 함수를 정의해주어야합니다.
        - upload_ocrbenchv2.py; custom dataset일 경우 HF에 업로드를 돕는 스크립트입니다. 다른 task들 참고해서 작성했지만, 사용하진 않습니다. 이미 HF에 올라와있는 dataset을 Load해서 사용하는 구조이기 때문입니다.
        - utils.py; 위의 yaml 파일에서 사용하는 함수들을 정의하는 스크립트입니다.

### CODE REVIEW

File: lmms_eval/models/docvision.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def get_config(pretrained):
             test=True,
         )
         self._model.eval()
-        self._eos_token_id = self.tokenizer.encode(self._eos_token)[1]
+        self._eos_token_id = self.tokenizer.convert_tokens_to_ids(self._eos_token)
```
Comment (hancheolcho) [inline lmms_eval/models/docvision.py:77]: 찾아보니 `encode()`는 tokenize & convert_tokens_to_ids 를 수행하는 함수군요.
여기선 변환할 토큰을 알고 있으니, 변경된 코드 (convert_tokens_to_ids를 직접 사용)가 더 적합한 것 같네요.

File: lmms_eval/models/docvision_utils/modeling_phi3.py
Comment (hancheolcho) [inline lmms_eval/models/docvision_utils/modeling_phi3.py:1]: modeling_phi3.py 추가된 것을 확인했습니다.

File: lmms_eval/models/docvision_utils/modeling_phi3.py [hunk 2]
```diff
@@ -0,0 +1,1702 @@
...
+#                 is_causal=self.is_causal,
+#             )
+
+#         attn_output = attn_output.reshape(bsz, q_len, self.hidden_size).contiguous()
+#         attn_output = self.o_proj(attn_output)
+
+#         if not output_attentions:
+#             attn_weights = None
+
+#         return attn_output, attn_weights, past_key_value
```
Comment (hancheolcho) [inline lmms_eval/models/docvision_utils/modeling_phi3.py:730]: 이게 슬랙에서 FA => SDPA 쓰면 돌아간다고 얘기하셨던 부분이군요.

File: lmms_eval/tasks/ocrbench_v2_en/requirements.txt
Comment (hancheolcho) [inline lmms_eval/tasks/ocrbench_v2_en/requirements.txt:1]: 오, task에 requirements.txt 도 줄 수 있군요.

File: lmms_eval/tasks/ocrbench_v2_en/ocrbenchv2.yaml
Comment (hancheolcho) [inline lmms_eval/tasks/ocrbench_v2_en/ocrbenchv2.yaml:1]: ocrbenchv2_en.yaml 이랑 내용이 똑같은데, 디폴트로 실행하게 하려고 해놓으신걸까요?

기존 태스크들을 보니 여러 sub-task가 있을 때에 최상단 task yaml 파일은 해당 sub-task들을 포함하는 그룹으로 지정하더라구요.

예, https://github.com/UpstageAI/lmms-eval/tree/main/lmms_eval/tasks/mmmu
image

ocrbenchv2도 위처럼 모든 sub-task를 다 돌리는 yaml 파일로 하거나, 혹은 디폴트 모드로 돌리는게 낫다면 아래처럼 해보면 어떨까요?

```yaml
group: ocrbenchv2
task:
- ocrbenchv2_en
```

File: lmms_eval/tasks/ocrbench_v2_en/IoUscore_metric.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    if "answer" in predict.keys():
+        score_content = vqa_evaluation(predict["answer"], img_metas["answers"])
+    if "bbox" in predict.keys():
+        gt_bbox = img_metas["bbox"]
+        try:
+            predict_bbox_list = ast.literal_eval(predict["bbox"])
+            score_bbox = calculate_iou(predict_bbox_list, gt_bbox)
+        except:
+            score_bbox = 0
+    return 0.5 * score_content + 0.5 * score_bbox
```
Comment (hancheolcho) [inline lmms_eval/tasks/ocrbench_v2_en/IoUscore_metric.py:44]: OcrBenchV2는 최종 점수 구할 때 content 랑 iou 스코어를 반반 하는군요.

File: lmms_eval/tasks/ocrbench_v2_en/TEDS_metric.py [hunk 4]
```diff
@@ -0,0 +1,930 @@
...
+        if (node1.tag != node2.tag) or (node1.colspan != node2.colspan) or (node1.rowspan != node2.rowspan):
+            return 1.
+        if node1.tag == 'td':
+            if node1.content or node2.content:
+                return self.normalized_distance(node1.content, node2.content)
+        return 0.
+
+
+class TEDS(object):
+    ''' Tree Edit Distance basead Similarity
```
Comment (hancheolcho) [inline lmms_eval/tasks/ocrbench_v2_en/TEDS_metric.py:74]: 오잉 오타가 있네요.
공식 레포에도 이 오타가 있군요 ㅋ

File: lmms_eval/tasks/ocrbench_v2_en/IoUscore_metric.py
Comment (hancheolcho) [inline lmms_eval/tasks/ocrbench_v2_en/IoUscore_metric.py:1]: OcrBenchV2 공식 레포와 동일한 파일들은 자세히 보지 않고 넘겼습니다.

File: lmms_eval/tasks/ocrbench_v2_en/upload_ocrbenchv2.py
Comment (hancheolcho) [inline lmms_eval/tasks/ocrbench_v2_en/upload_ocrbenchv2.py:1]: 이게 미팅 때 얘기해주셨던 내용 (데이터를 HF data hub 에 넣어야한다는)이군요.

**Review by hancheolcho — COMMENTED** [review]
OcrBenchV2 태스크 추가 고생 많으셨습니다.
공식 레포 기능을 이식하는데 쉽지 않았을 것 같네요.

궁금한 점은 코멘트를 남겼습니다.
답변 받는데로 Approve하겠습니다.

File: lmms_eval/tasks/ocrbench_v2_en/ocrbenchv2.yaml
Comment (seunggu-upstage) [reply lmms_eval/tasks/ocrbench_v2_en/ocrbenchv2.yaml:1]: en만 따로 올린다는게 같이 올라갔네요..!

말씀주신대로 en과 cn을 별도의 task로 뜯으려면 HF에 두 데이터 셋이 별도로 올라가있어야 하더라구요. 근데 우선 OCRBenchv2 저자들이 올린 HF dataset 원본을 사용하고자 해서 일단 하나로 동작하도록 구현했습니다!

**Review by hancheolcho — APPROVED** [review]
답변 확인했습니다.
고생하셨습니다~
